### ISSUE TITLE
Dashboard became a copy of another dashboard

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/frontend#20462]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Got a very crucial issue while attempting to test [another issue](https://github.com/home-assistant/frontend/issues/20460).
A whole storage-mode dashboard was replaced by another storage-mode dashboard - i.e. I got 2 (almost) same dashboards:

1. Assume dashboard_1 has 20 views, dashboard_2 has 10 views.
2. You need to move card_X from some view on dashboard_1 to some view of_dashboard_2:

![image](https://github.com/home-assistant/frontend/assets/71872483/42a9519a-11ce-4b15-ac46-29f2d100e0a5)

3. Then move this card back to the original view.

4. Result: dashboard_1 becomes SAME as dashboard_2 - only this card_X is a difference.
And a json file for this lost dashboard is overwritten with a content of another dashboard.
Here the 1st "lovelace.lovelace" (dashboard_1) file became a copy of the "lovelace.lovelace_css" (dashboard_2) (as I said - only the moved card_X is a difference):

![image](https://github.com/home-assistant/frontend/assets/71872483/462a975d-b48f-47bd-b1ad-e961d217145c)

What I tried to do:
1. Opened that json file of the overwritten dashboard (`sudo nano lovelace.lovelace`).
2. Pasted a content of the same file from a backup.
3. Cleared a browser's cache just in case.
4. But still the dashboard is shown as a copy of another dashboard.
5. Fixed only after rebooting HA.

### Describe the behavior you expected

see above

### Steps to reproduce the issue

as above

### What version of Home Assistant Core has the issue?

2024.4.1

### In which browser are you experiencing the issue with?

Chrome 123.0.6312.106

### Which operating system are you using to run this browser?

Win10x64

**Comment by ildar170975** [comment]
Cannot (& do not want to) reproduce. Closing.
